### DISCUSSION TITLE
GitLab Auth failing

### DISCUSSION THREAD

**Discussion by Chad-ll** [Kitware/CDash#2434] [Q&A] [answered]
Hi everyone,

I am setting up a brand new instance of CDash 3.5.0 and am having a bear of a time getting GitLab authentication to work. This server currently only has the admin user and no data.

For comparison, my other two existing CDash servers that were upgrades from 3.4.x -> 3.5.0 work fine with GitLab. These two servers had existing users/data.

When selecting GitLab as the authentication source, my CDash and GitLab servers will talk to each other, then CDash will say "500 Server Error" with the following in `cdash.log`:

> ```production. ERROR: Undefined array key 1 {"exception":[object] {ErrorException(code: 0): Undefined array key 1 at /opt/CDash/CDash-3.5.0/app/Http/Controllers/OAuthController.php:39) ```

My `.env` config
```
GITLAB_ENABLE=true
GITLAB_CLIENT_ID=<client id>
GITLAB_CLIENT_SECRET=<secret>
GITLAB_DOMAIN=https://gitlab.my.domain/
GITLAB_AUTO_REGISTER_NEW_USERS=true
GITLAB_REDIRECT_URI=https://cdash.my.domain/auth/gitlab/callback
```

GitLab is configured with:
* read_api
* read_user
* openid
* profile
* email

(More of a "please work" collection, since my other two instances are only using "read_user") We are also using LDAP and Kerberos as identity providers on GitLab. 

Running on RHEL 8.10, httpd/PHP 8.2. 

I fully expect that I'm doing something dumb and have just missed something obvious.

Thanks for any help.

**Comment by josephsnyder** [comment]
Hi @Chad-ll, thanks for the question! [The line that the error message is referencing](https://github.com/Kitware/CDash/blob/master/app/Http/Controllers/OAuthController.php#L39) deals with the name of the person coming back from GitLab. 

Does the "real" name of the GitLab account that you're signing into only have one name? Or no name set at all?  Without a name string with a space in it, it wouldn't `explode` into the first and last name variables that we expect.

**Reply by Chad-ll** [reply, marked as answer]
Hi @josephsnyder  

I went and looked and for whatever reason, all of our GitLab users are set with their login as their 'real name.' I just never noticed that before. 

After I changed my 'real name' field on GitLab, everything worked as expected with CDash. Yay!

Thank you for taking the time to explain what that code was doing and pointing me in the direction of the fix.

**Reply by josephsnyder** [reply]
Great! I think your original setup is probably fairly common.  I've created [an issue](https://github.com/Kitware/CDash/issues/2435) to not assume that `name` is a first and last name and we can let CDash account for it to avoid it being a 500 error
